internal/cmdrender/runtime/container.go

```diff
@@ -103,3 +126,34 @@ func (f *ContainerFn) getDockerCmd() (*exec.Cmd, context.CancelFunc) {
 	ctx, cancel := context.WithTimeout(context.Background(), timeout)
 	return exec.CommandContext(ctx, path, args...), cancel
 }
+
+// prepareImage will check local images and pull it if it doesn't
+// exist.
+func (f *ContainerFn) prepareImage() error {
```

Comment:
I don't remember the reason for doing this ? 
One thing worries me is that the options (timeout, docker auth etc.) for `docker image pull` may differ from the default (or user configured) settings.

Reply:
Because the command `docker run` will pull the missing images and the progress will be printed to stderr and pollute the output.

Reply:
Ack.